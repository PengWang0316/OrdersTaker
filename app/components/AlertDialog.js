import React from 'react';
import PropTypes from 'prop-types';
import { Dialog, DialogTitle, DialogContent, DialogContentText, DialogActions, Button } from '@material-ui/core';

export const AlertDialog = ({
  open, onClose, title, content, onSecondButton, onFirstButton, firstButtonText, secondButtonText
}) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
    <DialogContent>
      <DialogContentText id="alert-dialog-description">{content}</DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onFirstButton} color="primary">
        {firstButtonText}
      </Button>
      <Button onClick={onSecondButton} color="primary" autoFocus>
        {secondButtonText}
      </Button>
    </DialogActions>
  </Dialog>
);
AlertDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  onFirstButton: PropTypes.func.isRequired,
  onSecondButton: PropTypes.func.isRequired,
  firstButtonText: PropTypes.string,
  secondButtonText: PropTypes.string
};
AlertDialog.defaultProps = {
  firstButtonText: 'Cancel',
  secondButtonText: 'Confirm'
};
export default AlertDialog;
