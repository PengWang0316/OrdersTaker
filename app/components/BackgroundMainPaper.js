import React from 'react';
import PropTypes from 'prop-types';
import { withStyles, Paper } from '@material-ui/core';

const styles = {
  root: {
    backgroundImage: 'linear-gradient(to right top, #051937, #004d7a, #008793, #00bf72, #a8eb12)',
    width: '100%',
    height: '100%',
    padding: '0 20px'
  },
  paper: {
    margin: '40px auto 20px auto',
    maxWidth: 1024,
    borderRadius: '10px'
  }
};

export const BackgroundMainPaper = ({ classes, children }) => (
  <div className={classes.root}>
    <Paper className={classes.paper} elevation={4}>
      {children}
    </Paper>
  </div>
);
BackgroundMainPaper.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.element.isRequired
};
export default withStyles(styles)(BackgroundMainPaper);
