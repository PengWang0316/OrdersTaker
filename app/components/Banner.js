import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Hidden from '@material-ui/core/Hidden';
import { withStyles } from '@material-ui/core/styles';

import { fetchBasicInformation } from '../actions/BasicInformationActions';

const styles = {
  divPlaceHolder: {
    background: '#efefef',
    width: '100%',
    minHeight: '400px'
  },
  bigBanner: {
    width: '100%',
    maxHeight: '770px'
  },
  smallBanner: {
    width: '100%',
    maxHeight: '400px'
  },
  bannerText: {
    display: 'flex',
    flexDirection: 'column',
    width: '60%',
    position: 'absolute',
    marginTop: '18%',
    left: '20%'
  },
  title: {
    fontSize: '40px',
    color: 'white',
    textAlign: 'center',
    textTransform: 'uppercase',
    textShadow: '1px 1px 1px #909090'
  },
  subTitle: {
    color: 'white',
    fontSize: '24px',
    textAlign: 'center',
    textShadow: '1px 1px 1px #909090'
  },
  '@media (max-width: 600px)': {
    bannerText: { marginTop: '21%' },
    title: { fontSize: '30px' },
    subTitle: { fontSize: '18px' },
    divPlaceHolder: { minHeight: '230px' }
  }
};

/**
 * The Component to show the banner image and title.
 */
export class Banner extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    basicInformation: PropTypes.object,
    fetchBasicInformation: PropTypes.func.isRequired
  };
  static defaultProps = { basicInformation: null };

  /**
   * Call the redux action to get the basic infomariont.
   * @param {object} props contains this component's prop values;
   * @return {null} No return.
   */
  constructor(props) {
    super(props);
    if (!props.basicInformation) props.fetchBasicInformation();
  }

  /**
   * Render the jsx for the component.
   * @return {jsx} Return the jsx.
   */
  render() {
    const { classes, basicInformation } = this.props;
    return (
      <div className={basicInformation ? '' : classes.divPlaceHolder}>
        {basicInformation &&
          <Fragment>
            <div className={classes.bannerText}>
              <div className={classes.title}>{basicInformation.banners[0].title}</div>
              <div className={classes.subTitle}>{basicInformation.banners[0].subTitle}</div>
            </div>
            <Hidden only="xs">
              <img alt="Big Banner" src={basicInformation.banners[0].url} className={classes.bigBanner} />
            </Hidden>
            <Hidden only={['lg', 'md', 'sm']}>
              <img alt="Small Banner" src={basicInformation.banners[0].xsUrl} className={classes.smallBanner} />
            </Hidden>
          </Fragment>}
      </div>
    );
  }
}
/* istanbul ignore next */
const mapStateToProps = ({ basicInformation }) => ({ basicInformation });
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  fetchBasicInformation: () => dispatch(fetchBasicInformation())
});
export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Banner));
