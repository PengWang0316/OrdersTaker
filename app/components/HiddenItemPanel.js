import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import { ExpandMore, ExpandLess } from '@material-ui/icons';
import { scroller } from 'react-scroll';

import MenuItem from './MenuItem';

const flexBox = {
  flexWrap: 'wrap',
  height: 'auto',
  maxHeight: 0,
  // lineHeight: 0,
  display: 'flex',
  overflow: 'hidden',
  transition: 'max-height 0.7s'
};

const styles = {
  buttonDiv: {
    width: '100%',
    textAlign: 'right'
  },
  flexBoxAround: {
    ...flexBox,
    justifyContent: 'space-around'
  },
  // flexBoxStart: {
  //   ...flexBox,
  //   justifyContent: 'flex-start'
  // },
  '@media (max-width: 500px)': {
    flexBoxAround: { justifyContent: 'center' }
  }
};

/**
 * Show the loading button and extra items
 */
export class HiddenItemPanel extends Component {
  static propTypes = {
    items: PropTypes.array.isRequired,
    category: PropTypes.string.isRequired,
    // itemAmount: PropTypes.number.isRequired,
    classes: PropTypes.object.isRequired
  };

  state = { isExpand: false };

  /**
   * Setting the isExpand state to true and scroll to the button's position when a user click the button.
   * react-scroll library is used.
   * @return {null} No return.
   */
  handleShowMoreClick = () => {
    // this.categoryTitle = this.categoryTitle || document.getElementById(this.props.category);
    this.setState(({ isExpand }) => {
      // if (isExpand) this.categoryTitle.scrollIntoView();
      if (isExpand) scroller.scrollTo(this.props.category, {
        duration: 800,
        delay: 0,
        smooth: 'easeInOutQuart'
      });
      return { isExpand: !isExpand };
    });
  }

  /**
   * Rendering the jsx
   * @return {jsx} Return jsx.
   */
  render() {
    const { items, classes } = this.props;
    const { isExpand } = this.state;
    const divHeight = isExpand ? `${287 * (items.length + 1)}px` : 0; // Max height equal items' number times each element's height(287px).

    return (
      <Fragment>
        {items.length !== 0 && (
          <Fragment>
            <div className={classes.flexBoxAround} style={{ maxHeight: divHeight }}>
              {items.map(item => <MenuItem key={item._id} item={item} />)}
            </div>
            <div className={classes.buttonDiv}>
              <Button color="primary" onClick={this.handleShowMoreClick}>{isExpand ? (<Fragment>Hide <ExpandLess /></Fragment>) : (<Fragment>Show More <ExpandMore /></Fragment>)}</Button>
            </div>
          </Fragment>
        )}
      </Fragment>);
  }
}
export default withStyles(styles)(HiddenItemPanel);
