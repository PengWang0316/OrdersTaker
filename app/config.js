export const JWT_MESSAGE = 'OrdersTaker_JWT';
export const ADMINISTRATOR_ROLE = 1;
export const SUPER_USER_ROLE = 2;
export const NORMAL_USER_ROLE = 3;
// export const BASE_URL = 'https://orders-taker.glitch.me'; // Development server
// export const BASE_URL = 'https://orderstaker.kevin-project.com:8081'; // Development server on the AWS
export const BASE_URL = 'https://orderstaker.kevin-project.com:8080'; // Production server
// export const SOCKETIO_URL = 'https://orderstaker.kevin-project.com:3001';
export const SOCKETIO_URL = 'https://orderstaker.kevin-project.com:3000'; // Production server

export const CLOUDINARY_UPLOAD_PRESET = 'OrdersTakerWebApp'; // The preset of Cloudinary.
export const LOGIN_CALLBACK_URL = 'OrdersTaker_loginCallbackUrl';

export const ITEM_ONE_PRICE_KEY = '_onePrice';
export const MAX_ORDER_AMOUNT = 6; // How many orders will show in the one page
export const ORDER_STATUS_FINISHED = 'Finished';
export const ORDER_STATUS_RECEIVED = 'Received';
export const ORDER_STATUS_PREPARING = 'Preparing';

/* Lazy image loading */
export const LAZY_IMAGE_CLASS = 'lazyImage';
export const IMAGE_PLACEHOLDER_URL = 'https://res.cloudinary.com/orderstaker/image/upload/v1536616770/others/PlaceHolder.png';

/* URLS */
export const HOME_PAGE_URL = '/';
export const LOGIN_REDIRECT_RUL = '/loginRedirect';
export const CART_PAGE_URL = '/cartPage';
export const ORDERS_PAGE_URL = '/ordersPage';
export const KITHEN_INTERFACE_PAGE_URL = '/kithenInterfacePage';

/* Socket IO events */
export const SOCKETIO_EVENT_ADD_NEW_ORDER = 'socketioEventAddNewOrder';
export const SOCKETIO_EVENT_UPDATE_ORDER_ITEM = 'socketioEventUpdateOrderItem';
