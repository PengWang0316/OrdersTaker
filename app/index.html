<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="A simply order taking system for restaurants">
    <meta name="keywords" content="Order taker restaurant free">
    <!-- Add manifest -->
    <link rel="manifest" href="/manifest.json">
    <!-- Tell the browser it's a PWA -->
    <meta name="mobile-web-app-capable" content="yes">
    <!-- Tell iOS it's a PWA -->
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="theme-color" content="#6d9cb5"/>
    <link rel="icon" href="/favicon.png" >
    <!-- Latest compiled and minified CSS -->

    <title>Orders Taker - A simply order taking system for restaurants</title>
  </head>
  <body>
    <div id="app"></div>
    <noscript>This site requires JavaScript to work. Please change to a browser that supports JavaScript or turn on JavaScript in the setting.</noscript>
    <!-- Latest compiled and minified JavaScript -->
  </body>
</html>
