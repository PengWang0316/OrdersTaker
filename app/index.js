import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { AppContainer } from 'react-hot-loader';
import { PersistGate } from 'redux-persist/integration/react';

import App from './components/App';
import configureStore from './store/ConfigureStore';
import './styles/index.global.css';
import registerServiceWorker from './registerServiceWorker';

const { store, persistor } = configureStore();

// ReactDOM.render(
//   <Provider store={store}>
//     <App />
//   </Provider>,
//   document.getElementById('app')
// );
const render = Component => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <Component />
        </PersistGate>
      </Provider>
    </AppContainer>,
    document.getElementById('app'),
  );
};
render(App);

// Webpack Hot Module Replacement API
if (module.hot) module.hot.accept('./components/App', () => render(App));

// Registering a service worker in the production enviroment.
registerServiceWorker();
