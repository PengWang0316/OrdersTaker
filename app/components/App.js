
import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';
// import Loadable from 'react-loadable'; Does not work with react-hot-loader very well.
// import universal from 'react-universal-component'; Does not work with react-hot-loader very well.
import importedComponent from 'react-imported-component';
import CssBaseline from '@material-ui/core/CssBaseline';

import LoginDialog from './LoginDialog/';
import LoginDialogSnackbar from './snackbars/LoginDialogSnackbar';
import LogoutSnackbar from './snackbars/LogoutSnackbar';
import Navbar from './Navbar';
import { parserUserFromJwt } from '../actions/UserActions';
import LoginDialogContext from '../contexts/LoginDialogContext';

import { JWT_MESSAGE, HOME_PAGE_URL, LOGIN_REDIRECT_RUL, ORDER_PAGE_URL, ORDER_STATUS_PAGE_URL, ORDERS_PAGE_URL } from '../config';
import LoadingAnimation from './SharedComponents/LoadingAnimation';

// import { CSSTransitionGroup } from 'react-transition-group';

// import BackgroundMainPager from './BackgroundMainPager';

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#9ecde7',
      main: '#6d9cb5',
      dark: '#3e6e85',
      contrastText: '#fff',
    },
    secondary: {
      light: '#497ca4',
      main: '#105075',
      dark: '#002884',
      contrastText: '#fff',
    },
  },
});

/* istanbul ignore next */
const HomePage = importedComponent(() => import(/* webpackChunkName: "HomePageContainer" */ './containers/HomePageContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

/* istanbul ignore next */
const LoginRedirect = importedComponent(() => import(/* webpackChunkName: "LoginRedirect" */ './LoginRedirect').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

/* istanbul ignore next */
const OrderPage = importedComponent(() => import(/* webpackChunkName: "OrderPageContainer" */ './containers/OrderPageContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

/* istanbul ignore next */
const OrderStatusPage = importedComponent(() => import(/* webpackChunkName: "OrderStatusPageContainer" */ './containers/OrderStatusPageContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

/* istanbul ignore next */
const OrdersPage = importedComponent(() => import(/* webpackChunkName: "OrdersPageContainer" */ './containers/OrdersPageContainer').catch(err => console.log(err)), { LoadingComponent: LoadingAnimation });

/**
 * The root component that contains the theme, routers, navbar, and login dialog
 */
export class App extends Component {
  static propTypes = {
    user: PropTypes.object.isRequired,
    parserUserFromJwt: PropTypes.func.isRequired
  };

  /**
   * If Redux does not have user and local storage has jwt, use jwt to parser the user.
   * @param {object} props contains the component's prop value.
   * @return {null} No return.
   */
  constructor(props) {
    super(props);
    if (!props.user._id) {
      const jwtMessage = localStorage.getItem(JWT_MESSAGE);
      if (jwtMessage) props.parserUserFromJwt(jwtMessage);
    }
  }

  state = {
    isLoginDialogOpen: false,
    isLogoutSnackBarOpen: false,
    isLoginSnackbarOpen: false,
    snackbarMessage: ''
  };

  /**
   * Setting the open state to an opposite value.
   * @return {null} No return.
   */
  handleToggleDialog = () => this.setState(({ isLoginDialogOpen }) => ({ isLoginDialogOpen: !isLoginDialogOpen }));

  /**
   * Setting the snackbarOpen state to true in order to open the snackbar.
   * If message paramter is a string, set it to message state. Otherwise, set a empty message.
   * @param {string} message will be showed in the snackbar.
   * @return {null} No return.
   */
  handleToggleSnackbar = message => this.setState(({ isLoginSnackbarOpen }) =>
    ({ isLoginSnackbarOpen: !isLoginSnackbarOpen, snackbarMessage: typeof message === 'string' ? message : '' }));

  /**
   * Set the logoutSnackbarOpen state to an opposite value.
   * @return {null} No return.
   */
  handleToggleLogoutSnackbar = () => this.setState(({ isLogoutSnackBarOpen }) =>
    ({ isLogoutSnackBarOpen: !isLogoutSnackBarOpen }));

  /**
   * It will be pass down in a context in order to make it available for all component.
   * This method will be used when logout action is performed.
   * Clear the state isLoginDialogOpen in order to prevent the LoginDialog shows up.
   * @return {null} No return.
   */
  handleLogoutAction = () => this.setState({ isLoginDialogOpen: false, isLogoutSnackBarOpen: true });

  /**
   * The render method for the component
   * @return {jsx} Return jsx
   */
  render() {
    const { user } = this.props;
    const { isLogoutSnackBarOpen, isLoginDialogOpen, isLoginSnackbarOpen, snackbarMessage } = this.state;
    return (
      <MuiThemeProvider theme={theme}>
        <Router>
          {/* Using a context to give children components access to login dialog */}
          <LoginDialogContext.Provider value={{ handleLogoutAction: this.handleLogoutAction, handleToggleLoginDialog: this.handleToggleDialog }}>
            <div>
              <CssBaseline />
              <Navbar />
              <main>
                <Switch>
                  <Route exact path={HOME_PAGE_URL} component={HomePage} />
                  <Route exact path={ORDER_PAGE_URL} component={OrderPage} />
                  <Route exact path={LOGIN_REDIRECT_RUL} component={LoginRedirect} />
                  <Route path={`${ORDER_STATUS_PAGE_URL}/:orderId`} component={OrderStatusPage} />
                  <Route path={ORDERS_PAGE_URL} component={OrdersPage} />
                  <Route render={() => <p>Not Fount!</p>} />
                </Switch>
              </main>
              {!user._id && ( // If a user has already login, do not need to load componnets below.
                <Fragment>
                  <LoginDialog open={isLoginDialogOpen} onClose={this.handleToggleDialog} onToggleSnackbar={this.handleToggleSnackbar} />
                  <LoginDialogSnackbar open={isLoginSnackbarOpen} onClose={this.handleToggleSnackbar} message={snackbarMessage} />
                </Fragment>
              )}
              <LogoutSnackbar open={isLogoutSnackBarOpen} onClose={this.handleToggleLogoutSnackbar} />
            </div>
          </LoginDialogContext.Provider>
        </Router>
      </MuiThemeProvider>
    );
  }
}
/* istanbul ignore next */
const mapStateToProps = state => ({
  user: state.user
});
/* istanbul ignore next */
const mapDispatchToProps = dispatch => ({
  parserUserFromJwt: jwtMessage => dispatch(parserUserFromJwt(jwtMessage))
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
