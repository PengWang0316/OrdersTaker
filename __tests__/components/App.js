import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { App } from '../../app/components/App';
import { JWT_MESSAGE } from '../../app/config';

jest.mock('react-router-dom', () => ({
  BrowserRouter: ({ children }) => children,
  Route: ({ path, render }) => {
    if (render) return `<Route render=${render()} />`;
    return `<Route path=${path} />`;
  },
  Switch: ({ children }) => children
}));
jest.mock('../../app/components/Navbar', () => 'Navbar');
jest.mock('../../app/components/containers/HomePageContainer', () => 'HomePageContainer');
jest.mock('../../app/components/LoginDialog', () => 'LoginDialog');
jest.mock('../../app/components/snackbars/LoginDialogSnackbar', () => 'LoginDialogSnackbar');
jest.mock('../../app/components/snackbars/CheckCircleIconSnackbar', () => 'CheckCircleIconSnackbar');

window.console.log = jest.fn(); // Silencing the console log.

describe('App component test', () => {
  const defaultProps = {
    user: { _id: 'userId' },
    parserUserFromJwt: jest.fn()
  };
  const getShallowComponent = (props = defaultProps) => shallow(<App {...props} />);

  test('Constructor and initial states', () => {
    // localStorage.removeItem(JWT_MESSAGE);
    const { isLoginDialogOpen, isLogoutSnackBarOpen, isLoginSnackbarOpen, snackbarMessage } = getShallowComponent().state();
    expect(isLoginDialogOpen).toBe(false);
    expect(isLogoutSnackBarOpen).toBe(false);
    expect(isLoginSnackbarOpen).toBe(false);
    expect(snackbarMessage).toBe('');
    expect(defaultProps.parserUserFromJwt).not.toHaveBeenCalled();

    // getShallowComponent({ ...defaultProps, user: {} });
    // expect(defaultProps.parserUserFromJwt).not.toHaveBeenCalled();

    // localStorage.setItem(JWT_MESSAGE, 'message');
    // getShallowComponent({ ...defaultProps, user: {} });
    // expect(defaultProps.parserUserFromJwt).toHaveBeenCalledTimes(1);
    // expect(defaultProps.parserUserFromJwt).toHaveBeenLastCalledWith('message');
    // localStorage.removeItem(JWT_MESSAGE);
  });

  test('handleToggleDialog', () => {
    const component = getShallowComponent();
    expect(component.state('isLoginDialogOpen')).toBe(false);
    component.instance().handleToggleDialog();
    expect(component.state('isLoginDialogOpen')).toBe(true);
  });

  test('handleToggleSnackbar', () => {
    const component = getShallowComponent();
    expect(component.state('isLoginSnackbarOpen')).toBe(false);
    expect(component.state('snackbarMessage')).toBe('');

    component.instance().handleToggleSnackbar({});
    expect(component.state('isLoginSnackbarOpen')).toBe(true);
    expect(component.state('snackbarMessage')).toBe('');

    component.instance().handleToggleSnackbar('New message');
    expect(component.state('isLoginSnackbarOpen')).toBe(false);
    expect(component.state('snackbarMessage')).toBe('New message');
  });

  test('handleToggleLogoutSnackbar', () => {
    const component = getShallowComponent();
    expect(component.state('isLogoutSnackBarOpen')).toBe(false);
    component.instance().handleToggleLogoutSnackbar();
    expect(component.state('isLogoutSnackBarOpen')).toBe(true);
  });

  test('handleLogoutAction', () => {
    const component = getShallowComponent();
    component.instance().handleLogoutAction();
    expect(component.state('isLoginDialogOpen')).toBe(false);
    expect(component.state('isLogoutSnackBarOpen')).toBe(true);
  });

  test('App snapshot with user id', () => expect(renderer.create(<App {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('App snapshot without user id', () => expect(renderer.create(<App {...{ ...defaultProps, user: {} }} />).toJSON()).toMatchSnapshot());
});
