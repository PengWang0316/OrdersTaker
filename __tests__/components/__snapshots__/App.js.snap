// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test App snapshot with user id 1`] = `
<div>
  <Navbar />
  <main>
    &lt;Route path=/ /&gt;
    &lt;Route path=/orders /&gt;
    &lt;Route path=/loginRedirect /&gt;
    &lt;Route path=/orderStatus/:orderId /&gt;
    &lt;Route path=/ordersPage /&gt;
    &lt;Route render=[object Object] /&gt;
  </main>
  <LogoutSnackbar
    onClose={[Function]}
    open={false}
  />
</div>
`;

exports[`App component test App snapshot without user id 1`] = `
<div>
  <Navbar />
  <main>
    &lt;Route path=/ /&gt;
    &lt;Route path=/orders /&gt;
    &lt;Route path=/loginRedirect /&gt;
    &lt;Route path=/orderStatus/:orderId /&gt;
    &lt;Route path=/ordersPage /&gt;
    &lt;Route render=[object Object] /&gt;
  </main>
  <LoginDialog
    onClose={[Function]}
    onToggleSnackbar={[Function]}
    open={false}
  />
  <LoginDialogSnackbar
    message=""
    onClose={[Function]}
    open={false}
  />
  <LogoutSnackbar
    onClose={[Function]}
    open={false}
  />
</div>
`;
